### DISCUSSION TITLE
When using a chatbot with Gemini 1.5 Flush as a model, an error occurs from the second post with the same conversation ID.

### DISCUSSION THREAD

**Discussion by ijoko86** [langgenius/dify#9281] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

![Difyreport](https://github.com/user-attachments/assets/e7a4cac5-85e2-4659-8694-f46f05e10d3b)

In the cloud version of Dify v0.9.1, an error occurs when sending a second request for the same conversation ID of a chatbot using Gemini. This occurred in Gemini 1.5 Flash, but also in other models such as Gemini 1.5 Flush Latest. It did not occur in the OpenAI model.

### 2. Additional context or comments

I used the Dify Cloud Free plan to test this case.

**Comment by kurokobo** [comment]
It is probably a bug. The self-hosted environment with the image `langgenius/dify-api:0.9.1-fix1` does not have the issue, but this is reproduced with the current `langgenius/dify-api:main`.

I haven't carefully tracked the code yet, but it seems that when sending the second message, the `prompt_messages` passed to `GoogleLargeLanguageModel._generate()` include an empty message (from assistant) and duplicate messages (from user).

I ran it through the logger roughly, and this is what happened:

On the first message:

```python
role=<PromptMessageRole.SYSTEM: 'system'> content='Answer in English.\n' name=None
role=<PromptMessageRole.USER: 'user'> content='Hello!' name=None
```

On the second message:

```python
role=<PromptMessageRole.SYSTEM: 'system'> content='Answer in English.\n' name=None
role=<PromptMessageRole.USER: 'user'> content='Hello!' name=None
role=<PromptMessageRole.ASSISTANT: 'assistant'> content='Hello! 👋 How can I help you today? 😊' name=None tool_calls=[]
role=<PromptMessageRole.USER: 'user'> content='I just wanted to say hello :P' name=None
role=<PromptMessageRole.ASSISTANT: 'assistant'> content='' name=None tool_calls=[]     # 👈👈👈👈👈👈 empty message
role=<PromptMessageRole.USER: 'user'> content='I just wanted to say hello :P' name=None     # 👈👈👈👈👈👈 duplicated message
```

The last two lines did not exist in 0.9.1-fix1.
This empty message is being passed to Gemini, leading to an error.

https://github.com/langgenius/dify/blob/d97d3ff5fc43a066c605a04c4ac5d522155527dd/api/core/model_runtime/model_providers/google/llm/llm.py#L195-L200

https://github.com/langgenius/dify/blob/d97d3ff5fc43a066c605a04c4ac5d522155527dd/api/core/model_runtime/model_providers/google/llm/llm.py#L209-L215

**Comment by kurokobo** [comment]
https://github.com/langgenius/dify/issues/9296

**Comment by kurokobo** [comment]
https://github.com/langgenius/dify/pull/9297
